app/helpers/sufia/sufia_helper_behavior.rb

```diff
@@ -38,7 +38,7 @@ def sufia_thumbnail_tag(document, options)
       # file
       else
         path = if document.image? || document.pdf? || document.video? || document.office_document?
-          sufia.download_path document.noid, datastream_id: 'thumbnail'
+          sufia.download_path document, datastream_id: 'thumbnail'
         elsif document.audio?
```

Comment:
You're right, I'm relying on `#to_param` magic rather than explicitly specifying the `#id` method. (And whether you use UUIDs or NOIDs, `#to_param` will return the object's identifier.)